File: 백준/Silver/28278. 스택 2/스택 2.kt
import java.util.*

fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.out.bufferedWriter()
    val t = br.readLine().toInt()
    val stack = Stack<Int>()
    val st = StringBuilder()

    for (i in 0 until t) {
        val input = br.readLine()
        val args = input.split(" ")
        val commend = args.first()

        if (commend == "1") {
            stack.add(args[1].toInt())
            continue
        }

        val value = when (commend) {
            "2" -> if (stack.isEmpty()) -1 else stack.pop()
            "3" -> stack.size
            "4" -> if (stack.isEmpty()) 1 else 0
            "5" -> if (stack.isEmpty()) -1 else stack.peek()
            else -> throw IllegalArgumentException()
        }
        st.append(value).append('\n')
    }
    bw.write(st.toString())
    br.close()
    bw.close()
}

File: 백준/Silver/28278. 스택 2/스택 2.py
import sys

t = int(sys.stdin.readline())
stack = []

for _ in range(t):
    args = list(map(int, sys.stdin.readline().split(" ")))
    cmd = args[0]

    if cmd == 1:
        stack.append(int(args[1]))
    elif cmd == 2:
        print(-1 if not stack else stack.pop())
    elif cmd == 3:
        print(len(stack))
    elif cmd == 4:
        print(1 if not stack else 0)
    elif cmd == 5:
        print(-1 if not stack else stack[-1])
